javascript/test/GherkinInMarkdownTokenMatcherTest.ts

```diff
@@ -34,35 +34,58 @@ describe('GherkinInMarkdownTokenMatcher', function () {
     assert.strictEqual(token.matchedText, 'hello')
   })
 
+  it('matches FeatureLine without the Feature: keyword', () => {
+    const line = new GherkinLine('# hello', location.line)
+    const token = new Token(line, location)
+    assert(tm.match_FeatureLine(token))
+    assert.strictEqual(token.matchedType, TokenType.FeatureLine)
+    assert.strictEqual(token.matchedKeyword, undefined)
+    assert.strictEqual(token.matchedText, '# hello')
+  })
+
   it('matches bullet Step', () => {
     const line = new GherkinLine('  *  Given I have 3 cukes', location.line)
     const token = new Token(line, location)
     assert(tm.match_StepLine(token))
     assert.strictEqual(token.matchedType, TokenType.StepLine)
     assert.strictEqual(token.matchedKeyword, 'Given ')
+    assert.strictEqual(token.matchedKeywordType, 'Context')
     assert.strictEqual(token.matchedText, 'I have 3 cukes')
     assert.strictEqual(token.location.column, 6)
   })
-
+  
   it('matches plus Step', () => {
     const line = new GherkinLine('  +  Given I have 3 cukes', location.line)
     const token = new Token(line, location)
     assert(tm.match_StepLine(token))
     assert.strictEqual(token.matchedType, TokenType.StepLine)
     assert.strictEqual(token.matchedKeyword, 'Given ')
+    assert.strictEqual(token.matchedKeywordType, 'Context')
     assert.strictEqual(token.matchedText, 'I have 3 cukes')
     assert.strictEqual(token.location.column, 6)
   })
-
+  
   it('matches hyphen Step', () => {
     const line = new GherkinLine('  -  Given I have 3 cukes', location.line)
     const token = new Token(line, location)
     assert(tm.match_StepLine(token))
     assert.strictEqual(token.matchedType, TokenType.StepLine)
     assert.strictEqual(token.matchedKeyword, 'Given ')
+    assert.strictEqual(token.matchedKeywordType, 'Context')
     assert.strictEqual(token.matchedText, 'I have 3 cukes')
     assert.strictEqual(token.location.column, 6)
   })
+  
+  it('matches a when Step', () => {
```

Comment:
I added this test as I had a bug in my Python parser.

Is this an appropriate location to be testing `matchedKeywordType`

If these are too low level (white-box / implementation detail), then the tests should be moved/removed